core/domain/skill_services.py

```diff
@@ -514,3 +515,107 @@ def save_skill_summary(skill_summary):
+        skill_id: str. The unique id of the skill.
+        degree_of_mastery: float. The degree of mastery of user in the skill.
+    """
+    model_id = user_models.UserSkillMasteryModel.get_model_id(
+        user_id, skill_id)
+
+    user_skill_mastery = skill_domain.UserSkillMastery(
+        model_id, user_id, skill_id, degree_of_mastery)
+    save_user_skill_mastery(user_skill_mastery)
+
+
+def save_user_skill_mastery(user_skill_mastery):
+    """Stores skill mastery of a user.
+
+    Args:
+        user_skill_mastery: dict. The user skill mastery model of a user.
+    """
+    user_skill_mastery_model = user_models.UserSkillMasteryModel(
+        id=user_skill_mastery.id,
+        user_id=user_skill_mastery.user_id,
+        skill_id=user_skill_mastery.skill_id,
+        degree_of_mastery=user_skill_mastery.degree_of_mastery)
+
+    user_skill_mastery_model.put()
+
+
+def get_skill_mastery(user_id, skill_id):
+    """Fetches the mastery of user in a particular skill.
+
+    Args:
+        user_id: str. User id of the user currently logged in.
+        skill_id: str. Unique id of the skill for which mastery degree is
+            requested.
+
+    Returns:
+        degree_of_mastery: float. Mastery degree of the user for the
+            requested skill.
+    """
+    degree_of_mastery = user_models.UserSkillMasteryModel.get(
+        user_models.UserSkillMasteryModel.get_model_id(
+            user_id, skill_id)).degree_of_mastery
+
+    return degree_of_mastery
+
+
+def get_multi_skill_mastery(user_id, skill_ids):
+    """Fetches the mastery of user in multiple skills.
+
+    Args:
+        user_id: str. User id of the user currently logged in.
+        skill_ids: list(str). Skill IDs of the skill for which mastery degree is
+            requested.
+
+    Returns:
+        degree_of_mastery: list(float). Mastery degree of the user for requested
+            skills.
+    """
+    degrees_of_mastery = []
+    model_ids = []
+
+    for skill_id in skill_ids:
+        model_ids.append(
+            user_models.UserSkillMasteryModel.get_model_id(user_id, skill_id))
+
+    skill_mastery_models = user_models.UserSkillMasteryModel.get_multi(
+        model_ids)
+    for skill_summary in skill_mastery_models:
+        degrees_of_mastery.append(skill_summary.degree_of_mastery)
+
+    return degrees_of_mastery
+
+
+def get_mastery_for_all_skills(user_id):
+    """Fetches the mastery of user for all skills.
+
+    Args:
+        user_id: str. User id of the user currently logged in.
+
+    Returns:
+        skill_mastery: dict. Mastery degree of the user for every skill with
+            skill_id as key and corresponding degree of mastery of user
+            as value.
+            Format: { "str": float }
+            Eg: skill_mastery = {
+                "ph57nkgc4" : 1.0,
+                "zds64cgfv" : 0.1,
+                "gxcs52kug" : 0.0,
+            }
+    """
+    user_skill_mastery = {}
```

Comment:
skill_ids_to_degrees_of_mastery would probably give the reader a better idea for what the variable contains and how it's structured.

More generally though -- is this a construct you plan to use? If so we should have a domain object for it, rather than passing around raw dicts.

Reply:
No, this variable is just to return the output.

Reply:
But is a controller going to call this? If so, make a domain object for the return value, and that domain object should have a to_dict() method. If not, why have this function at all?